File: couples/python/free_man.py
class FreeMan:
    def __init__(self, id, next):
        self.id = id
        self.next = next

    def check(self):
        free_man = self
        while (free_man):
            #print(free_man.id)
            free_man = free_man.next
    
    def add(self, man_id):
        free_man = self
        while (free_man.next):
            free_man = free_man.next
        free_man.next = FreeMan(man_id, None)

    @staticmethod
    def get_free_men_list(quantity):
        old_free_man = None
        new_free_man = None
        
        for i in range(0, quantity):
            old_free_man = new_free_man
            new_free_man = FreeMan(i, old_free_man)

        return new_free_man

File: couples/java/src/zeionara/couples/FreeMan.java
package zeionara.couples;

public class FreeMan{
	private int id;
	private FreeMan next;

	public int get_id(){
		return id;
	}

	public FreeMan get_next(){
		return next;
	}

	public void set_next(FreeMan free_man){
		next = free_man;
	}
	
	public FreeMan(int id, FreeMan next){
		this.id = id;
		this.next = next;
	}

	public static FreeMan getFreeMenList(int quantity){
		FreeMan oldFreeMan = null;
		FreeMan newFreeMan = null;

		for (int i = 0; i < quantity; i++){
			oldFreeMan = newFreeMan;
			newFreeMan = new FreeMan(i, oldFreeMan);
		}
		
		return newFreeMan;
	}

	public void add(int man_id){
		FreeMan free_man = this;

		while (free_man.get_next() != null){
			free_man = free_man.get_next();
		}

		free_man.set_next(new FreeMan(man_id, null));
	}

	public void check(){
		FreeMan freeMan = this;

		while(freeMan != null){
			System.out.println(freeMan.id);
			freeMan = freeMan.next;
		}
	}
}

